src/node/txreconciliation.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <node/txreconciliation.h>
+
+#include <util/system.h>
+
+#include <unordered_map>
+#include <variant>
+
+
+namespace {
+
+/**
+ * Keeps track of reconciliation-related per-peer state.
+ */
+class ReconciliationState
+{
+};
+
+} // namespace
+
+/** Actual implementation for TxReconciliationTracker's data structure. */
+class TxReconciliationTracker::Impl
+{
+    mutable Mutex m_mutex;
+
+    /**
+     * Keeps track of reconciliation states of eligible peers.
+     * For pre-registered peers, the locally generated salt is stored.
+     * For registered peers, the locally generated salt is forgotten, and the state (including
+     * "full" salt) is stored instead.
+     */
+    std::unordered_map<NodeId, std::variant<uint64_t, ReconciliationState>> m_states GUARDED_BY(m_mutex);
+
+public:
+    // Local protocol version
+    // Made public to supress -Wunused-private-field. Should be made private when becomes used.
+    const uint32_t m_recon_version;
```

Comment:
https://github.com/bitcoin/bitcoin/commit/0d7823a20650ef3b5f2c2da6b33f6dd142636c38 It might be simpler to add this data member when it is first used, which avoids needing to move it and delete the comments at that point.

Reply:
My compiler gives this:

```
node/txreconciliation.cpp:29:20: warning: private field 'm_recon_version' is not used [-Wunused-private-field]
    const uint32_t m_recon_version;
```

Reply:
So i'll stick to the first option you suggest i guess.